injector/src/main/java/edu/ucr/cs/riple/injector/changes/ChangeVisitor.java

```diff
@@ -0,0 +1,183 @@
+    }
+    return ans.get();
+  }
+
+  @Override
+  @Nullable
+  public Modification visitField(OnField onField, Pair<NodeList<BodyDeclaration<?>>, Change> pair) {
+    final AtomicReference<Modification> ans = new AtomicReference<>();
+    final NodeList<BodyDeclaration<?>> members = pair.a;
+    final Change change = pair.b;
+    members.forEach(
+        bodyDeclaration ->
+            bodyDeclaration.ifFieldDeclaration(
+                fieldDeclaration -> {
+                  if (ans.get() != null) {
+                    // already found the member.
+                    return;
+                  }
+                  NodeList<VariableDeclarator> vars =
+                      fieldDeclaration.asFieldDeclaration().getVariables();
+                  for (VariableDeclarator v : vars) {
+                    if (onField.variables.contains(v.getName().toString())) {
+                      ans.set(change.visit(fieldDeclaration));
+                      break;
+                    }
+                  }
+                }));
+    return ans.get();
+  }
+
+  @Override
+  @Nullable
+  public Modification visitParameter(
+      OnParameter onParameter, Pair<NodeList<BodyDeclaration<?>>, Change> pair) {
+    final AtomicReference<Modification> ans = new AtomicReference<>();
+    final NodeList<BodyDeclaration<?>> members = pair.a;
+    final Change change = pair.b;
+    members.forEach(
+        bodyDeclaration ->
+            bodyDeclaration.ifCallableDeclaration(
+                callableDeclaration -> {
+                  if (ans.get() != null) {
+                    // already found the member.
+                    return;
+                  }
+                  if (onParameter.matchesCallableDeclaration(callableDeclaration)) {
+                    NodeList<?> params = callableDeclaration.getParameters();
+                    if (onParameter.index < params.size()) {
+                      if (params.get(onParameter.index) != null) {
+                        Node param = params.get(onParameter.index);
+                        if (param instanceof Parameter) {
+                          ans.set(change.visit((Parameter) param));
+                        }
+                      }
+                    }
+                  }
+                }));
+    return ans.get();
+  }
+
+  @Override
+  @Nullable
+  public Modification visitClass(OnClass onClass, Pair<NodeList<BodyDeclaration<?>>, Change> pair) {
+    final NodeList<BodyDeclaration<?>> members = pair.a;
+    final Change change = pair.b;
+    if (isAnonymousClassFlatName(change.location.clazz)) {
+      return null;
+    }
+    // Get the enclosing class of the members
+    Optional<Node> optionalClass = members.getParentNode();
+    if (optionalClass.isEmpty() || !(optionalClass.get() instanceof BodyDeclaration<?>)) {
+      return null;
+    }
+    return change.visit(((BodyDeclaration<?>) optionalClass.get()));
+  }
+
+  /**
+   * Applies the change to the compilation unit.
```

Comment:
This doesn't actually apply any change, right?  It just computes the change (`Modification`) to be applied?

Reply:
Yes exactly. I reworded the java doc to mention this, please let me know if I should update it. 2e2dd6b83be323d4f97964abe8749cb242564aa8

Reply:
Hrm, I don't see the javadoc changes here; maybe the commit got lost somehow?

Reply:
Sorry for that, the change was on the javadoc for the class. Update the method javadoc as well 66edb7ea867ee787ae68f7ebe2a367b2e3bc8e5c